webapp/src/components/lhs/group.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {useState} from 'react';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import {Box, Button, List, Typography, ButtonProps} from '@mui/material';
+import {styled} from '@mui/material/styles';
+
+import {SidebarGroup, InProgressNodesCategoryName, NextNodesCategoryName} from '../../types/sidebar';
+import {Client} from '../../client/client';
+
+import Item from './item';
+
+
+interface GroupProps {
+    group: SidebarGroup;
+}
+
+const Group = (props: GroupProps) => {
+    const [collapsed, setCollapsed] = useState(props.group.collapsed);
+
+    return (
+        <Box
+            boxSizing={'border-box'}
+            color={'primary.contrastText'}
+        >
+            <HeaderButton
+                variant='text'
+                startIcon={collapsed? <ChevronRightIcon/> :<ExpandMoreIcon/>}
+                aria-label={props.group.display_name}
+                onClick={() => {
+                    // Currently InProgressNodes category and NextNodes category are automatically generated and
+                    // not saved in the DB. So we can't yet save the collapse state for these categories.
+                    if (props.group.id !== InProgressNodesCategoryName && props.group.id !== NextNodesCategoryName) {
+                        Client.Sidebar().setMyCategoryCollapsed(props.group.id, !collapsed);
+                    }
+                    setCollapsed(!collapsed);
+                }}
+            >
+                <Typography
+                    width={'100%'}
+                    fontSize={14}
+                    fontWeight={600}
+                >
+                    {props.group.display_name}
+                </Typography>
+            </HeaderButton>
+            <List dense={true}>
```

Comment:
Would it be better to include the `List` component only when the group is not collapsed?

Reply:
good point